### PR TITLE
Dialog::Marketplace() : Fix usage of wrong kingdom during battle (if called from surrender dialog)

### LINKED ISSUE #3633: Combat, when the hero surrenders, the gold amounts do not match at the Marketplace
In fheroes2, when the hero surrenders, the gold amounts do not match at the Marketplace.

In fheroes2:

Gold Marketplace bug flèche 01

https://user-images.githubusercontent.com/43583994/120905170-88d0ca00-c650-11eb-8b80-f50a3785f96c.mp4

End the turn.

Here is the save file:
[Gold Marketplace bug 01.zip](https://github.com/ihhub/fheroes2/files/6603223/Gold.Marketplace.bug.01.zip)

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #3633

### CODE REVIEW

File: src/fheroes2/dialog/dialog_marketplace.cpp [hunk 1]
```diff
@@ -437,11 +438,11 @@ void Dialog::Marketplace( Kingdom & kingdom, bool fromTradingPost )
 
                 if ( resourceTo )
                     cursorTo.hide();
-                RedrawToResource( pt2, true, fromTradingPost, resourceFrom );
+                RedrawToResource( pt2, true, kingdom, fromTradingPost, resourceFrom );
                 if ( resourceTo )
                     cursorTo.show();
                 if ( resourceTo )
```
Comment (felix642) [inline src/fheroes2/dialog/dialog_marketplace.cpp:444]: I know this was already like this, but do you really need to check the same condition twice ?

Comment (ihhub) [reply src/fheroes2/dialog/dialog_marketplace.cpp:444]: Hi @felix642 , you have a good point. It will lead for a cleaner and faster code after all

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for the fix!

File: src/fheroes2/dialog/dialog_marketplace.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/fheroes2/dialog/dialog_marketplace.cpp]: Yes, didn't notice it :)

**Pull request merged**